AK/StringView.h

```diff
@@ -42,6 +42,29 @@ class StringView {
         , m_length(bytes.size())
     {
     }
+#if __has_builtin(__builtin_strnlen)
+    template<typename T, std::size_t S>
+    ALWAYS_INLINE StringView(T (&characters)[S])
+        : m_characters((const char*)characters)
+        , m_length(__builtin_strnlen(characters, S))
+    {
+        VERIFY(!Checked<uintptr_t>::addition_would_overflow((uintptr_t)characters, m_length));
+    }
+#else
+    template<typename T, std::size_t S>
+    ALWAYS_INLINE StringView(T (&characters)[S])
+        : m_characters((const char*)characters)
+        , m_length(S - 1)
+    {
+        for (std::size_t i = 0; i < S; ++i) {
```

Comment:
pls no `std::type`, I'm surprised that even works.
`size_t` is already defined at this point, please use that instead.

Reply:
I'll need some clarification around this feedback. size_t and std::size_t are functionally the same but size_t is the C way and std::size_t is the C++ way. The style guide says to write c++ rather than c so why is the recommendation here to do it the c way? If this is too long of a conversation for here I'm on the discord server.

Reply:
`size_t` is the preferred type used in the project.
in general, the STL and its defines are avoided in favour of AK and whatever we've defined there.